llvm/unittests/TargetParser/TargetParserTest.cpp

```diff
@@ -163,6 +163,15 @@ template <typename T> struct ARMCPUTestParams {
     return os;
   }
 
+  static std::string
+  PrintToStringParamName(const testing::TestParamInfo<ARMCPUTestParams<T>>& Info) {
+    std::string Name = Info.param.CPUName.str();
+    for (char &C : Name)
+      if (!std::isalnum(C))
```

Comment:
Does this ever happen? Are there CPU names that have weird chars in it? If yes, why can't we just print them?

Reply:
Oh it can't print "cortex-whatever", it needs to replace "-" with "_"?

How strange...

Reply:
A bunch of them have `-`s in them, and it is a requirement that gtest test names only have alphanumeric characters in them: https://github.com/google/googletest/blob/main/docs/advanced.md#specifying-names-for-value-parameterized-test-parameters

Reply:
I suspect it has to do with simplifying command-line argument parsing.

Reply:
Worth adding a comment mentioning the requirement from the suite